public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -162,16 +183,10 @@ export function getPendingPeriod(rule: CombinedRule): string | undefined {
   return undefined;
 }
 
-export function getAnnotations(rule: CombinedRule): Annotations | undefined {
-  if (
-    isRecordingRulerRule(rule.rulerRule) ||
-    isRecordingRule(rule.promRule) ||
-    isGrafanaRecordingRule(rule.rulerRule)
-  ) {
-    return undefined;
-  }
-  return rule.annotations ?? [];
+export function getAnnotations(rule?: Rule): Annotations | undefined {
```

Comment:
It feels like this function never returns undefined

Reply:
I think I'll update this to just only accept `AlertingRule` and make the caller check for `prometheusRuleType.alertingRule()`